Library/Homebrew/os/mac/cache_store.rb

```diff
@@ -0,0 +1,207 @@
+require "dbm"
+require "json"
+
+#
+# `DatabaseCache` is a class acting as an interface to a persistent storage
+# mechanism residing in the `HOMEBREW_CACHE`
+#
+class DatabaseCache
+  # Name of the database cache file located at <HOMEBREW_CACHE>/<name>.db
+  #
+  # @return [String]
+  attr_accessor :name
+
+  # Instantiates new `DatabaseCache` object
+  #
+  # @param  [String] name
+  # @return [nil]
```

Comment:
I'm not convinced that comments these structured are helpful. It's expected that `initialize` will always instantiate a new object and return `nil`. I think in general it's worth trimming down the comments to explaining why things are done rather than what. Comments should be a last resort when the code cannot adequately communicate the same information.

Reply:
That's a fair point.

I can remove the comments if you'd like. I had them to be compliant with RDoc specifications. I personally like being explicit about what types parameters and return values should be.

Reply:
@AndrewMcBurney I think it's reasonable to document parameters and return types but only add additional text if it says something that the name doesn't.

Reply:
Alright, sounds good to me! 🙂